### ISSUE TITLE
Project Document Logic

### ISSUE THREAD

**Issue by jonscottfilm** [karmaradio/karma#436]
Document-upload UX simplification, and routing logic:

2 dropdowns:

**1st dropdown:** 
_(and in this order in all views too please. Explainer notes in brackets to appear in dropdown list)_

CONTRACTS (deal-related agreements)
FORMS (paperwork everyone must sign)*
INFORMATION (read-only, sent to all)

**2nd dropdown:** (dependent on first)

_Contracts:_ 
(anything in bold shows in dropdown)
**PAYE** (cross-ref job title list)
**Schedule-D** (cross-ref job title list)
**Direct Hire (combined PAYE & Schedule-D)**
**Loan Out** (conditional: 3 x loan out company fields completed)
**Construction PAYE** (cross-ref _department and_ job title list)
**Construction Schedule-D** (cross-ref _department and_ job title list)
**Construction Direct Hire (combined PAYE & Schedule-D)**
**Construction Loan Out** (conditional: 3 x loan out company fields completed)
**Transport PAYE** (cross-ref _department and_ job title list)
**Transport Schedule-D** (cross-ref _department and_ job title list)
**Transport Direct Hire (combined PAYE & Schedule-D)**
**Transport Loan Out** (conditional: 3 x loan out company fields completed)
**Box Rental** (you know the rest)
**Equipment Rental
Vehicle Allowance**

**Instead of 2nd dropdown**, if FORMS or INFORMATION is selected, offer a free-text field as such:

_Forms:_ (no logic required, all forms (with document specific approval chain), sent to everyone)
Label: Document name: 
Field: [ user custom filename               ]
Button: UPLOAD
Upload as many as project requires

 _Information:_ (no logic required, all information visible to everyone)
Label: Document name: 
Field: [ user custom filename               ]
Button: UPLOAD
Upload as many as project requires

**Comment by JMurphyWeb** [comment]
@jonscottfilm moving this out of Sprint 4 for now.

**Comment by jonscottfilm** [comment]
What was the upshot of how @jackcarlisle proceeds, regarding this?

**Comment by jackcarlisle** [comment]
@jonscottfilm so I've finished the implementation for how it is currently working for all documents. I'm moving on to this method now with the text inputs for Info and Forms

**Comment by jonscottfilm** [comment]
gotya, thanks Jack.

**Comment by jackcarlisle** [comment]
@jonscottfilm did you get a chance to find the document where the information is held about what departments and job titles results in the differing contracts

**Comment by JMurphyWeb** [comment]
@jackcarlisle check the docs on drive shared with you?

**Comment by JMurphyWeb** [comment]
I just shared it

**Comment by jackcarlisle** [comment]
Add daily contracts

**Comment by jonscottfilm** [comment]
Will test now

**Comment by jonscottfilm** [comment]
Seems to be working!

I actually now need to methodically test the logic in ALL those document types.... oof.

**Comment by jonscottfilm** [comment]
@jackcarlisle - Does this include the conditional logic for PAYE grades having equipment becoming Schedule-D?

Also (make a new issue :)) - need to discuss an **over-ride** button in the PM offer creation view, to over-ride as such: PAYE to Sched-D, and vice versa. i.e. REVERSES whatever the conditional logic is from the job title table. (in order to revise an offer, prior to contractor accepting)

**Comment by jackcarlisle** [comment]
@jonscottfilm no this doesn't have the equipment logic or the premises logic for Schedule-D. Could you please open another issue and prioritise?

**Comment by jonscottfilm** [comment]
Yes, will do. I imagine the schedule-d/equipment logic, is slightly easier to implement than the over-ride button?

**Comment by jackcarlisle** [comment]
@jonscottfilm yeah and it's a nicer solution because the logic version will be automatic

**Comment by jonscottfilm** [comment]
OK. Having said that, the over-ride button, would be a solve-all...

**Comment by jackcarlisle** [comment]
@jonscottfilm sorry I meant that the other method was a nicer solution, not the over-ride button

**Comment by jonscottfilm** [comment]
@jackcarlisle - I know you did... :) If you can do the simpler, PAYE+equipment=Sched-D, that would be great for MVP.

**Comment by jonscottfilm** [comment]
Just a reminder of this logic please: https://github.com/karmaradio/karma/issues/436#issuecomment-313380293

Along with NOT showing Loan Out in the offer, by default. (not sure if/where that lives in an issue)
